modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -328,20 +341,34 @@ public CompletableFuture<Void> dropZone(String name) {
             CompletableFuture<Void> fut = new CompletableFuture<>();
 
             zonesConfiguration.change(zonesChange -> zonesChange.changeDistributionZones(zonesListChange -> {
-                DistributionZoneView view = zonesListChange.get(name);
+                DistributionZoneView zoneView = zonesListChange.get(name);
 
-                if (view == null) {
+                if (zoneView == null) {
                     throw new DistributionZoneNotFoundException(name);
                 }
 
+                NamedConfigurationTree<TableConfiguration, TableView, TableChange> tables = tablesConfiguration.tables();
+
+                boolean bindTable = tables.value().namedListKeys().stream()
```

Comment:
The idea was that this is not possible, since we do this in the "change". But it turned out that it doesn't work that way. I'll create a ticket for this issue, but for now let's leave it like that.